### PR TITLE
added versioned promotion and grant getters and test

### LINKED ISSUE #326: include claim protocol version in promotion object
for https://github.com/brave/browser-laptop/issues/14401 to be implemented we need to send the protocol version in the promotion offer object. I would suggest the following:
1. add version to any new promotion objects
1. we bump the promotion endpoint version, the v2 endpoint will filter for promotions with a version number and the legacy v1 endpoint will filter on promotions without a version number

### PR DESCRIPTION
**Pull request by maikelmclauflin**
fixes #326

### CODE REVIEW

**Comment by maikelmclauflin** [comment]
@evq are there any other endpoints that need updating at this time or are we just doing the gets for now

File: ledger/controllers/grants.js [hunk 1]
```diff
@@ -380,12 +430,19 @@ const uploadGrants = function (runtime) {
...
       state = {
+        $set,
         $currentDate: { timestamp: { $type: 'timestamp' } },
-        $set: underscore.omit(entry, ['promotionId']),
         $inc: { count: promotionCounts[entry.promotionId] }
       }
-      await promotions.update({ promotionId: entry.promotionId }, state, { upsert: true })
+      await promotions.update({
+        promotionId: entry.promotionId,
+        protocolVersion
```
Comment (evq) [inline ledger/controllers/grants.js]: I don't think we should allow for the case where there are two documents with the same `promotionId` but different `protocolVersion`s

File: .travis.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ env:
     - "GRANT_REDIS_HOST=localhost"
     - "GRANT_SIGNATOR_PUBLIC_KEY=bbe028009e744422bfc39b0e226633324d8c5fe0744c87c1018b88a488bc5899"
     - "GRANT_WALLET_CARD_ID=f042845f-fa62-4022-8117-a476ec583a7b"
-    - "MONGODB_URI=mongodb://mongo/test"
+    - "MONGODB_URI=mongodb://mongo/ledger"
```
Comment (evq) [inline .travis.yml]: can you remove this env var declaration?

File: ledger/controllers/grants.js [hunk 3]
```diff
@@ -590,11 +649,14 @@ module.exports.initialize = async (debug, runtime) => {
...
-        timestamp: bson.Timestamp.ZERO
+        timestamp: bson.Timestamp.ZERO,
+
+        protocolVersion: 1
       },
       unique: [ { promotionId: 1 } ],
       others: [ { active: 1 }, { count: 1 },
-                { batchId: 1 }, { timestamp: 1 } ]
+                { batchId: 1 }, { timestamp: 1 },
+                { version: 1 } ]
```
[the remark this answers is not in the document]
Comment (maikelmclauflin) [reply ledger/controllers/grants.js]: good catch. thanks!

**Pull request merged**